ecs-cli/modules/cli/configure/migrate.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2015-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package configure
+
+import (
+	"html/template"
+	"io/ioutil"
+	"os"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+)
+
+func migrateWarning(cliConfig *config.CLIConfig) error {
+	var oldConfig string
+	dest, err := config.NewDefaultDestination()
+	if err != nil {
+		return err
+	}
+	dat, err := ioutil.ReadFile(config.ConfigFilePath(dest))
+	if err != nil {
+		return err
+	}
+	oldConfig = string(dat)
+
+	// format template
+	data := map[string]interface{}{
+		"OldConfig":                oldConfig,
+		"Cluster":                  cliConfig.Cluster,
+		"Region":                   cliConfig.Region,
+		"CFNStackName":             cliConfig.CFNStackName,
+		"ComposeServiceNamePrefix": cliConfig.ComposeServiceNamePrefix,
+		"AWSAccessKey":             cliConfig.AWSAccessKey,
+		"AWSSecretKey":             cliConfig.AWSSecretKey,
+	}
+
+	t := template.Must(template.New("message").Parse(messageTemplate))
+	if err := t.Execute(os.Stdout, data); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+var messageTemplate = `
+Old config
+-----------------------------------------------------
+~/.ecs/config
+{{.OldConfig}}
+
+New configs
+-----------------------------------------------------
+~/.ecs/config
+version: v1
+default: default
+cluster-configurations:
+  default:
+  - cluster: {{.Cluster}}
+  - region: {{.Region}}
+  - cfn-stack-name: {{.CFNStackName}}
+  - compose-service-name-prefix: {{.ComposeServiceNamePrefix}}
```

Comment:
`cfn-stack-name` and `compose-service-name-prefix` do not always exist.

Reply:
So unfortunately I'm having trouble with the template making it not show these if they are not found. Here's what I tried:
```
cluster-configurations:
  default:
    cluster: {{.Cluster}}
    region: {{.Region}}
{{if .CFNStackName}}
    cfn-stack-name: {{.CFNStackName}}
{{end}}
{{if .ComposeServiceNamePrefix}}    
    compose-service-name-prefix: {{.ComposeServiceNamePrefix}}
{{end}}
```

Unfortunately, for some reason that results in this output with random extra newlines:
```
cluster-configurations:
  default:
    cluster: INSERT-CLUSTER-NAME-HERE
    region: us-east-2

    cfn-stack-name: amazon-ecs-cli-setup-INSERT-CLUSTER-NAME-HERE

    - compose-service-name-prefix: ecscompose-service-

```

Reply:
So I ran `ecs-cli configure --cluster cli-demo --region us-west-2` and got this config:
```
version: v1
default: default
clusters:
  default:
    cluster: cli-demo
    region: us-west-2
    compose-service-name-prefix: ""
    cfn-stack-name: ""
```
So basically because of how the yaml library works, even empty fields will be present in the config. I don't think this is a problem, and I also think its therefore fine to show them explicitly as empty in the migrate command output.